From: Mike Kelly <pioto@pioto.org>
Date: Thu, 2 Apr 2009 15:16:46 +0000 (-0400)
Subject: Fix _filedir on bash 4.
X-Git-Url: http://git.debian.org/?p=bash-completion%2Fbash-completion.git;a=commitdiff_plain;h=1421e55aac075e13491cd212b796bdd453214a2c

Fix _filedir on bash 4.

We don't need to double-quote things for compgen w/ bash 4. Just putting
each file in double quotes is sufficient.

See: http://lists.gnu.org/archive/html/bug-bash/2009-03/msg00155.html
---

diff --git a/bash_completion b/bash_completion
index 0246cd8..4d21cf8 100644
--- a/bash_completion
+++ b/bash_completion
@@ -209,6 +209,12 @@ quote()
 # results in the original argument
 quote_readline()
 {
+	if [[ "${BASH_VERSINFO[0]}" -ge 4 ]] ; then
+		# This function isn't really necessary on bash 4
+		# See: http://lists.gnu.org/archive/html/bug-bash/2009-03/msg00155.html
+		echo "${1}"
+		return
+	fi
 	local t="${1//\\/\\\\}"
 	echo \'${t//\'/\'\\\'\'}\' #'# Help vim syntax highlighting
 }
